### PR TITLE
Rerun CMake when files are added/removed.

### PR DESCRIPTION
**Pull request by damonkohler**

### CODE REVIEW

File: cmake/functions.cmake [hunk 1]
```diff
@@ -106,6 +106,18 @@ macro(google_initialize_cartographer_project)
 
   message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
 
+  # Add a hook that reruns CMake when source files are added or removed.
+  set(LIST_FILES_CMD "find ${CMAKE_SOURCE_DIR}/ -not -iwholename '*.git*' | sort | sed 's/^/#/'")
+  set(FILES_LIST_PATH "${CMAKE_BINARY_DIR}/AllFiles.cmake")
+  set(DETECT_CHANGES_CMD "bash" "-c" "${LIST_FILES_CMD} | diff -N -q ${FILES_LIST_PATH} - || ${LIST_FILES_CMD} > ${FILES_LIST_PATH}")
```
Comment (SirVer) [inline cmake/functions.cmake]: break line

File: cmake/functions.cmake [hunk 2]
```diff
@@ -106,6 +106,18 @@ macro(google_initialize_cartographer_project)
...
+  add_custom_target(regen ALL
```
Comment (SirVer) [inline cmake/functions.cmake]: use a name that is never conflicting with some cmake target.

File: cmake/functions.cmake (hunk 1 above)
Comment (damonkohler) [inline cmake/functions.cmake]: Not possible. Requires CMake > 3.0

https://cmake.org/cmake/help/v3.0/manual/cmake-language.7.html#quoted-argument

File: cmake/functions.cmake (hunk 2 above)
Comment (damonkohler) [inline cmake/functions.cmake]: Done.

**Pull request merged**